File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/IntiharGirisimiKrizTespitVeriSetiFormViewModel.ts
//$A38EF9FC
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { IntiharGirisimKrizVeriSeti } from "NebulaClient/Model/AtlasClientModel";
import { IntiharGirisimPsikiyatTani } from "NebulaClient/Model/AtlasClientModel";
import { IntiharGirisimiYontemi } from "NebulaClient/Model/AtlasClientModel";
import { IntiharGirisimiVakaSonucu } from "NebulaClient/Model/AtlasClientModel";
import { IntiharGirisimiKrizNedeni } from "NebulaClient/Model/AtlasClientModel";
import { SKRSICD } from "NebulaClient/Model/AtlasClientModel";
import { SKRSPsikiyatrikTedaviGecmisi } from "NebulaClient/Model/AtlasClientModel";
import { SKRSIntiharKrizVakaTuru } from "NebulaClient/Model/AtlasClientModel";
import { SKRSIntiharGirisimiGecmisi } from "NebulaClient/Model/AtlasClientModel";
import { SKRSAilesindePsikiyatrikVaka } from "NebulaClient/Model/AtlasClientModel";
import { SKRSAilesindeIntiharGirisimi } from "NebulaClient/Model/AtlasClientModel";
import { SKRSIntiharKrizVakaSonucu } from "NebulaClient/Model/AtlasClientModel";
import { SKRSIntiharGirisimiyadaKrizNedenleri } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class IntiharGirisimiKrizTespitVeriSetiFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.IntiharGirisimiKrizTespitVeriSetiFormViewModel, Core';
    @Type(() => IntiharGirisimKrizVeriSeti)
    public _IntiharGirisimKrizVeriSeti: IntiharGirisimKrizVeriSeti = new IntiharGirisimKrizVeriSeti();
    @Type(() => IntiharGirisimPsikiyatTani)
    public IntiharGirisimPsikiyatTaniGridList: Array<IntiharGirisimPsikiyatTani> = new Array<IntiharGirisimPsikiyatTani>();
    @Type(() => IntiharGirisimiYontemi)
    public IntiharGirisimiYontemiGridList: Array<IntiharGirisimiYontemi> = new Array<IntiharGirisimiYontemi>();
    @Type(() => IntiharGirisimiVakaSonucu)
    public IntiharGirisimiVakaSonucuGridList: Array<IntiharGirisimiVakaSonucu> = new Array<IntiharGirisimiVakaSonucu>();
    @Type(() => IntiharGirisimiKrizNedeni)
    public IntiharGirisimiKrizNedeniGridList: Array<IntiharGirisimiKrizNedeni> = new Array<IntiharGirisimiKrizNedeni>();
    @Type(() => SKRSICD)
    public SKRSICDs: Array<SKRSICD> = new Array<SKRSICD>();
    @Type(() => SKRSPsikiyatrikTedaviGecmisi)
    public SKRSPsikiyatrikTedaviGecmisis: Array<SKRSPsikiyatrikTedaviGecmisi> = new Array<SKRSPsikiyatrikTedaviGecmisi>();
    @Type(() => SKRSIntiharKrizVakaTuru)
    public SKRSIntiharKrizVakaTurus: Array<SKRSIntiharKrizVakaTuru> = new Array<SKRSIntiharKrizVakaTuru>();
    @Type(() => SKRSIntiharGirisimiGecmisi)
    public SKRSIntiharGirisimiGecmisis: Array<SKRSIntiharGirisimiGecmisi> = new Array<SKRSIntiharGirisimiGecmisi>();
    @Type(() => SKRSAilesindePsikiyatrikVaka)
    public SKRSAilesindePsikiyatrikVakas: Array<SKRSAilesindePsikiyatrikVaka> = new Array<SKRSAilesindePsikiyatrikVaka>();
    @Type(() => SKRSAilesindeIntiharGirisimi)
    public SKRSAilesindeIntiharGirisimis: Array<SKRSAilesindeIntiharGirisimi> = new Array<SKRSAilesindeIntiharGirisimi>();
    @Type(() => SKRSIntiharKrizVakaSonucu)
    public SKRSIntiharKrizVakaSonucus: Array<SKRSIntiharKrizVakaSonucu> = new Array<SKRSIntiharKrizVakaSonucu>();
    @Type(() => SKRSIntiharGirisimiyadaKrizNedenleri)
    public SKRSIntiharGirisimiyadaKrizNedenleris: Array<SKRSIntiharGirisimiyadaKrizNedenleri> = new Array<SKRSIntiharGirisimiyadaKrizNedenleri>();
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/IntiharGirisimiKrizTespitVeriSetiFormViewModel.cs
//$7F924788
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Infrastructure.Helpers;

namespace Core.Controllers
{
    public partial class IntiharGirisimKrizVeriSetiServiceController
    {
        partial void PreScript_IntiharGirisimiKrizTespitVeriSetiForm(IntiharGirisimiKrizTespitVeriSetiFormViewModel viewModel, IntiharGirisimKrizVeriSeti intiharGirisimKrizVeriSeti, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionID = viewModel.GetSelectedEpisodeActionID();
            if (selectedEpisodeActionID.HasValue)
            {
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionID.Value);
                viewModel._IntiharGirisimKrizVeriSeti.EpisodeAction = episodeAction;
                viewModel._IntiharGirisimKrizVeriSeti.SubEpisode = episodeAction.SubEpisode;

            }
            ContextToViewModel(viewModel, objectContext);
        }

    }
}

namespace Core.Models
{
    public partial class IntiharGirisimiKrizTespitVeriSetiFormViewModel: BaseViewModel, IENabizViewModel
    {
        public void AddENabizObjectViewModelToContext(TTObjectContext objectContext)
        {
            objectContext.AddObject(this._IntiharGirisimKrizVeriSeti);

            if (this.IntiharGirisimiKrizNedeniGridList != null)   // Ek nesneler için
            {
                foreach (var item in this.IntiharGirisimiKrizNedeniGridList)
                {
                    var intiharGirisimiKrizNedeni = (IntiharGirisimiKrizNedeni)objectContext.AddObject(item);
                    intiharGirisimiKrizNedeni.IntiharGirisimKrizVeriSeti = this._IntiharGirisimKrizVeriSeti;
                }
            }

            if (this.IntiharGirisimiVakaSonucuGridList != null)   // Ek nesneler için
            {
                foreach (var item in this.IntiharGirisimiVakaSonucuGridList)
                {
                    var intiharGirisimiVakaSonucu = (IntiharGirisimiVakaSonucu)objectContext.AddObject(item);
                    intiharGirisimiVakaSonucu.IntiharGirisimKrizVeriSeti = this._IntiharGirisimKrizVeriSeti;
                }
            }

            if (this.IntiharGirisimiYontemiGridList != null)   // Ek nesneler için
            {
                foreach (var item in this.IntiharGirisimiYontemiGridList)
                {
                    var intiharGirisimiYontemi = (IntiharGirisimiYontemi)objectContext.AddObject(item);
                    intiharGirisimiYontemi.IntiharGirisimKrizVeriSeti = this._IntiharGirisimKrizVeriSeti;
                }
            }

            if (this.IntiharGirisimPsikiyatTaniGridList != null)   // Ek nesneler için
            {
                foreach (var item in this.IntiharGirisimPsikiyatTaniGridList)
                {
                    var intiharGirisimPsikiyatTani = (IntiharGirisimPsikiyatTani)objectContext.AddObject(item);
                    intiharGirisimPsikiyatTani.IntiharGirisimKrizVeriSeti = this._IntiharGirisimKrizVeriSeti;
                }
            }


            var intiharGirisimKrizVeriSeti = (IntiharGirisimKrizVeriSeti)objectContext.GetLoadedObject(this._IntiharGirisimKrizVeriSeti.ObjectID);


            new SendToENabiz(objectContext, intiharGirisimKrizVeriSeti.SubEpisode, intiharGirisimKrizVeriSeti.ObjectID, intiharGirisimKrizVeriSeti.ObjectDef.Name, "230", Common.RecTime());
        }
    }
}
